rotkehlchen/chain/evm/transactions.py

```diff
@@ -456,19 +457,45 @@ def get_or_query_transaction_receipt(self, tx_hash: EVMTxHash) -> 'EvmTxReceipt'
             )
 
         if len(result) == 0:
-            transaction = self.evm_inquirer.get_transaction_by_hash(tx_hash)
-            with self.database.user_write() as write_cursor:
-                dbevmtx.add_evm_transactions(write_cursor, [transaction], relevant_address=None)
-            period = TimestampOrBlockRange(
-                range_type='blocks',
-                from_value=transaction.block_number,
-                to_value=transaction.block_number,
-            )
-            if transaction.to_address is not None:  # internal transactions only through contracts
-                self._query_and_save_internal_transactions_for_range(
-                    address=None,  # get all internal transactions for the parent hash
-                    period=period,
+            if tx_hash == GENESIS_HASH:
+                # Query genesis transactions
+                with self.database.conn.read_ctx() as cursor:
+                    accounts_data = self.database.get_blockchain_account_data(
+                        cursor,
+                        self.evm_inquirer.chain_id.to_blockchain(),
+                    )
+                for data in accounts_data:
+                    self._query_and_save_transactions_for_range(
+                        address=data.address,
+                        period=TimestampOrBlockRange(range_type='blocks', from_value=0, to_value=0),  # noqa: E501
+                    )
+
+                # Check whether the genesis tx was added
+                with self.database.conn.read_ctx() as cursor:
+                    added_tx = dbevmtx.get_evm_transactions(
```

Comment:
Hmmm why do this check? It makes no sense.

Also .... what happens if you do write the genesis hash in the DB and then you add another address later, that also has a genesis transaction?

Reply:
> Hmmm why do this check? It makes no sense.

This check tells us whether any of the tracked addresses had a genesis transaction.

> Also .... what happens if you do write the genesis hash in the DB and then you add another address later, that also has a genesis transaction?

You will have to pass through the classic process of querying transactions (the one that happens on the Evm Transactions page in the app) and get the transactions queried for that address.

Reply:
> This check tells us whether any of the tracked addresses had a genesis transaction.

And? Why do you need this here? I don't; see it used

Reply:
As you can see in the check below ```python
if len(added_tx) == 0:
                    raise InputError(
                        f'There is no tracked {str(self.evm_inquirer.chain_id)} address that '
                        f'would have a genesis transaction',
                    )
```

if case if there will are no genesis transactions found, an input error is thrown

Reply:
In case of "normal" transactions if the transaction is not found, a remote error is raised.

So for genesis transactions we also need some error to be thrown.

Reply:
When can this happen? Can you describe the flow here? This function is called and handles remote error only at `decode_transaction_hashes`. So when can it get here with the genesis hash and no tracked address having a transaction?

Reply:
`decode_transaction_hashes` method is exposed via API, so anything can be passed there at any time.

However in the app it probably won't happen since for redecoding frontend sends only already existing transactions.

But if we consider only backend then this zero genesis hash can be sent to the backend at any time.